remote-api/src/main/java/org/alfresco/rest/framework/resource/parameters/where/QueryHelper.java

```diff
@@ -286,4 +304,180 @@ public static String stripQuotes(String toBeStripped)
         }
         return toBeStripped; //default to return the String unchanged.
     }
+
+    public static QueryResolver.WalkerSpecifier resolve(final Query query)
+    {
+        return new QueryResolver.WalkerSpecifier(query);
+    }
+
+    /**
+     * Helper class allowing WHERE query resolving using query walker. By default {@link BasicQueryWalker} is used, but different walker can be supplied.
+     */
+    public static abstract class QueryResolver<S extends QueryResolver<?>>

```

Comment:
The level of class nesting is a bit of an overkill to me here but don't mind my opinion too much as I may not be getting the whole picture right.

Reply:
IMO nested helper classes are meant to use only within `QueryResolver` using builder pattern, so they don't have to be placed outside it. In general I agree with you that in different use case it would be an overkill.